src/accelerate/accelerator.py

```diff
@@ -1702,6 +1702,13 @@ def _prepare_deepspeed(self, *args):
         deepspeed_plugin = self.deepspeed_plugin
 
         is_dataloader_present = any(isinstance(obj, torch.utils.data.DataLoader) for obj in args)
+        tp_size = self.deepspeed_plugin.deepspeed_config["tensor_parallel"].get("autotp_size", 0)
+        if tp_size > 1:
+            from torch.distributed.device_mesh import init_device_mesh
+
+            mesh_dim_name = "tp"
+            self.state.ds_device_mesh = init_device_mesh(self.device.type, (tp_size,), mesh_dim_names=(mesh_dim_name,))
```

Comment:
Let's make sure to guard this under the DS version which adds this feature in + the one for `torch.distributed.device_mesh` please :)